x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/rollup/action/TransportPutRollupJobAction.java

```diff
@@ -339,6 +367,32 @@ public void onTimeout(TimeValue timeout) {
         );
     }
 
+    private static boolean hasRollupIndices(ClusterState state) throws IOException {
+        // Sniffing logic instead of invoking sourceAsMap(), which would materialize the entire mapping as map of maps.
+        for (var imd : state.metadata()) {
+            if (imd.mapping() == null) {
+                continue;
+            }
+
+            try (var parser = XContentHelper.createParser(PARSER_CONFIGURATION, imd.mapping().source().compressedReference())) {
+                if (parser.nextToken() == XContentParser.Token.START_OBJECT) {
+                    if ("_doc".equals(parser.nextFieldName())) {
+                        if (parser.nextToken() == XContentParser.Token.START_OBJECT) {
+                            if ("_meta".equals(parser.nextFieldName())) {
+                                if (parser.nextToken() == XContentParser.Token.START_OBJECT) {
+                                    if ("_rollup".equals(parser.nextFieldName())) {
```

Comment:
> This looks pretty fragile, what if there's another field before _rollup?

I don't there will be another field. The logic in `TransportPutRollupJobAction#createMappings(...)` always add a `_doc._meta._rollup` field and the `PARSER_CONFIGURATION` constant in `TransportPutRollupJobAction` only allows one field to be present. Let me add some unit tests for this.

>  I'd go with using sourceAsMap.

Mappings can be pretty large and I like to avoid converting large mappings in map of maps in the jvm heap space.

Reply:
In this case, it should throw if there's another field?

Sorry, did you add a unittest this that I missed?

Reply:
Right, I don't see a case with multiple fields under `_doc._meta` in this file?

Reply:
Ah, I see. I added that now too: 3b2b145afb494d0fb644d7bd08887a7cf5ba908b